File: v1/ecmascript/sum-of-multiples/sum-of-multiples.js
export default function SumOfMultiples(multiples) {
    return {
        to: function(max) {
            let sum = 0;

            for (let i = 1; i < max; i++) {
                if (multiples.find(m => i % m === 0))
                    sum += i;
            }

            return sum;
        }
    }
}

File: prolog/sum-of-multiples/sum_of_multiples.pl
multiple(Factors, N) :- member(Factor, Factors), N mod Factor =:= 0, !.

sum_of_multiples(Factors, Limit, Sum) :-
    Max is Limit - 1,
    aggregate(sum(N), N^(between(1, Max, N), multiple(Factors, N)), Sum), !.
sum_of_multiples(_, _, 0).
